src/tagmanager/tm_ctags.c

```diff
@@ -120,7 +120,15 @@ static gboolean init_tag(TMTag *tag, TMSourceFile *file, const tagEntryInfo *tag
 	tag->name = g_strdup(tag_entry->name);
 	tag->type = type;
 	tag->local = tag_entry->isFileScope;
-	tag->pointerOrder = 0;	/* backward compatibility (use var_type instead) */
+	tag->flags = tm_tag_flag_none_t;
+	if (isTagExtraBitMarked(tag_entry, XTAG_ANONYMOUS))
+	{
+		tag->flags |= tm_tag_flag_anon_t;
+		/* Temporarily store kind letter for anon name generation to flags
+		 * as we don't have any other place where to store it. This will
+		 * be cleared later in rename_anon_tags() */
+		tag->flags |= (guint)kind_letter << 16;
```

Comment:
Can't you just add a member to `TMTag` instead of this workaround?

Reply:
I wasn't sure if it was OK to add new members to TMTag when it's exposed to plugins. I guess ABI should be bumped in this case? 

But I think having the original ctags kind letter would be useful to have in TMTag because once we map it to our `TMTagType` representation, there's no way back to get the original kind. This might be needed if we decide to generate tags files in the ctags format.

Reply:
I think it was discussed elsewhere, appending members is generally OK without bumping the ABI.

But I wouldn't fear ABI bumps anyway. For most users those go unnoticed because they just get their recompiled plugins from their distro, it's just a minor inconvenience for those who compile the plugins themselves.

Reply:
@kugel- is right, add the members to the end and its not necessary to bump the ABI.  

The ABI problem really only manifests itself when a distro releases Geany with a new ABI, but not the plugins at the same time (Debian has done that before).

Reply:
Since then the Debian package implements a technical barrier, where plugins depend on the exact ABI version (so in practice its not possible to update the geany package to a conflicting version, without also updating the plugins).